lib/Platform/ElasticSearchPlatform.php

```diff
@@ -388,44 +388,34 @@ private function cleanHost(string $host): string {
 	 */
 	private function connectToElastic(array $hosts) {
 
-		try {
-			$hosts = array_map([$this, 'cleanHost'], $hosts);
-			[$user, $pass] = $this->getBasicAuth($hosts);
-			$cb = ClientBuilder::create()
-							   ->setHosts($hosts)
-							   ->setRetries(3)
-							   ->setBasicAuthentication($user, $pass);
-			$this->client = $cb->build();
-
-//		}
-//		catch (CouldNotConnectToHost $e) {
-//			$this 'CouldNotConnectToHost';
-//			$previous = $e->getPrevious();
-//			if ($previous instanceof MaxRetriesException) {
-//				echo "Max retries!";
-//			}
-		} catch (Exception $e) {
-			throw $e;
-//			echo ' ElasticSearchPlatform::load() Exception --- ' . $e->getMessage() . "\n";
-		}
-	}
+        try {
+            $hosts = array_map([$this, 'cleanHost'], $hosts);
+            $cb = ClientBuilder::create()
+                ->setHosts($hosts)
+                ->setRetries(3);
+            $this->configureAuthentication($cb, $hosts);
+
+            $this->client = $cb->build();
+        } catch (Exception $e) {
```

Comment:
Would it not be better to catch for `Throwable`? (While in this case it does not matter as it is thrown right again).

Reply:
you're right; removed the useless catch here and there